UnitySDK/Assets/ML-Agents/Examples/3DBall/Scripts/Ball3DAgent.cs

```diff
@@ -63,7 +68,20 @@ public override void AgentReset()
         ballRb.velocity = new Vector3(0f, 0f, 0f);
         ball.transform.position = new Vector3(Random.Range(-1.5f, 1.5f), 4f, Random.Range(-1.5f, 1.5f))
                                       + gameObject.transform.position;
+        //Reset the parameters when the Agent is reset.
+        SetResetParameters();
+    }
 
+    public void SetBall()
+    {
+        //Set the attributes of the ball by fetching the information from the academy
+        ballRb.mass = resetParams["mass"];
+        scale = resetParams["scale"];
+        ball.transform.localScale = new Vector3(scale, scale, scale);
```

Comment:
Did we settle on changing the scale along all 3 axis or was there talk about changing just along one dimension ?

Reply:
Oval ball could be interesting

Reply:
I tried changing the dimensions of the ball along the y and z axis, however the results aren't as interesting as we would've hoped. Once the ball lies on its longer flat side, then the plate is pretty much static. There might be more factors that I would have to look at to make this possible.